receiver/sqlserverreceiver/scraper.go

```diff
@@ -0,0 +1,298 @@
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/winperfcounters"
+	windowsapi "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/winperfcounters"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlserverreceiver/internal/metadata"
+)
+
+type sqlServerScraper struct {
+	logger   *zap.Logger
+	config   *Config
+	watchers []winperfcounters.PerfCounterWatcher
+}
+
+// newSqlServerScraper returns a new sqlServerScraper.
+func newSqlServerScraper(logger *zap.Logger, cfg *Config) *sqlServerScraper {
+	return &sqlServerScraper{logger: logger, config: cfg}
+}
+
+// start creates and sets the watchers for the scraper.
+func (s *sqlServerScraper) start(ctx context.Context, host component.Host) error {
+	watchers := []winperfcounters.PerfCounterWatcher{}
+	for _, objCfg := range createWatcherConfigs() {
+		objWatchers, err := objCfg.BuildPaths()
+		if err != nil {
+			s.logger.Warn("some performance counters could not be initialized", zap.Error(err))
+		}
+		for _, objWatcher := range objWatchers {
+			watchers = append(watchers, objWatcher)
+		}
+	}
+	s.watchers = watchers
+
+	return nil
+}
+
+// scrape collects windows performance counter data from all watchers and then records/emits it using the metricBuilder
+func (s *sqlServerScraper) scrape(ctx context.Context) (pdata.Metrics, error) {
+	metricsByDatabase, errs := createMetricGroupPerDatabase(s.watchers)
+
+	metricsBuilder := metadata.NewMetricsBuilder(s.config.Metrics)
+
+	for key, metricGroup := range metricsByDatabase {
+		emitMetricGroup(metricsBuilder, metricGroup, key)
+	}
+
+	return metricsBuilder.Emit(), errs
+}
+
+func createMetricGroupPerDatabase(watchers []windowsapi.PerfCounterWatcher) (map[string][]winperfcounters.CounterValue, error) {
+	var errs error
+
+	metricsByDatabase := map[string][]winperfcounters.CounterValue{}
+	for _, watcher := range watchers {
+		counterValues, err := watcher.ScrapeData()
+		if err != nil {
+			errs = multierr.Append(errs, err)
+			continue
+		}
+		for _, counterValue := range counterValues {
+			key := counterValue.Attributes["instance"]
+
+			if metricsByDatabase[key] == nil {
+				metricsByDatabase[key] = []winperfcounters.CounterValue{counterValue}
+			} else {
+				metricsByDatabase[key] = append(metricsByDatabase[key], counterValue)
+			}
+		}
+	}
+
+	return metricsByDatabase, errs
+}
+
+func emitMetricGroup(metricsBuilder *metadata.MetricsBuilder, metricGroup []winperfcounters.CounterValue, databaseName string) {
+	now := pdata.NewTimestampFromTime(time.Now())
+
+	for _, metric := range metricGroup {
+		metricsBuilder.RecordAnyDataPoint(now, metric.Value, metric.MetricRep.Name, metric.MetricRep.Attributes)
+	}
+
+	if len(databaseName) > 0 {
```

Comment:
nit: This is the idiomatic way to check for an empty string in go
```suggestion
	if databaseName != 0 {
```